src/OpenTelemetry.AutoInstrumentation/Instrumentation.cs

```diff
@@ -70,6 +70,7 @@ public static void Initialize()
                 // Register to shutdown events
                 AppDomain.CurrentDomain.ProcessExit += OnExit;
                 AppDomain.CurrentDomain.DomainUnload += OnExit;
+                AppDomain.CurrentDomain.UnhandledException += OnUnhandledException;
```

Comment:
Using this functionality typically improves one set of problems (losing extra telemetry) but creates a new set of problems (apps that hang instead of crash, apps that obscure the error in the user's code with errors from OT's cleanup code). It might still be a very favorable trade if 10% of the time the results are improved vs 0.1% of the time the results are worse, it just something to be aware of and be pragmatic about.